transformer/tf/model.py

```diff
@@ -0,0 +1,59 @@
+import tensorflow as tf
+
+def attention(q, k, v, mask=None, dropout=None):
+    "Compute 'Scaled Dot Product Attention'"
+
+    dk = tf.cast(tf.shape(k)[-1], tf.float32)
+
+    matmul_qk = tf.matmul(q, k, transpose_b=True)
+    scaled_attention_logits = matmul_qk / tf.math.sqrt(dk)
+
+    if mask is not None:
+        scaled_attention_logits += (mask * -1e9)
```

Comment:
i think theres a bug here. you want to override `scaled_attention_logits` to be large negative number (-1e9) wherever mask is "true". Currently you are doing a += which doesnt do this exactly. Would recommend to use tf.where:
```
negatives = tf.ones_like(logits) * -1e9
logits = tf.where(mask, negatives, logits)
```

Reply:
I raised `NotImplimentedError` for the mask. 
As I tackle this part I'll add tests and logic accordinly.